sdk/eventhub/azure-eventhub/tests/perfstress_tests/receive_event.py

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ._test_base import _ReceiveTest
+
+
+class ReceiveEventTest(_ReceiveTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        self._total_received_cnt = 0
+        self._async_total_received_cnt = 0
+
+    def _on_event_received(self, partition_context, event):
+        self._total_received_cnt += 1
+        if self._total_received_cnt > self.args.num_events:
+            self.consumer.close()
+
+    async def _on_event_received_async(self, partition_context, event):
+        self._async_total_received_cnt += 1
+        if self._async_total_received_cnt > self.args.num_events:
+            await self.async_consumer.close()
+
+    def run_sync(self):
+        self._total_received_cnt = 0
```

Comment:
this can move to setup - no?

Reply:
My understanding is that each time after we finish the `run` method, the status related to each running shall be reset -- in the EH case, the received cnt shall be cleaned up.
well, this is some overhead introduced by the push callback mode  -- another choice is to use a nested method for callback so that we could keep every status inside the method scope, but it further complicates the run method, so I'd rather to keep it as it is for now to make it work first and improve in the future.

Reply:
Yes I think this needs to be here. Otherwise the counts will continue to accumulate between runs.